app/src/main/java/io/novafoundation/nova/app/root/navigation/navigators/deepLinking/DeepLinkingNavigator.kt

```diff
@@ -1,36 +1,41 @@
 package io.novafoundation.nova.app.root.navigation.navigators.deepLinking
 
+import io.novafoundation.nova.app.R
+import io.novafoundation.nova.app.root.navigation.navigators.BaseNavigator
+import io.novafoundation.nova.app.root.navigation.navigators.NavigationHoldersRegistry
+import io.novafoundation.nova.app.root.navigation.openSplitScreenWithInstantAction
 import io.novafoundation.nova.feature_account_api.presenatation.account.add.ImportAccountPayload
 import io.novafoundation.nova.feature_account_impl.presentation.AccountRouter
 import io.novafoundation.nova.feature_assets.presentation.AssetsRouter
-import io.novafoundation.nova.feature_dapp_impl.presentation.DAppRouter
+import io.novafoundation.nova.feature_assets.presentation.balance.detail.BalanceDetailFragment
 import io.novafoundation.nova.feature_dapp_api.presentation.browser.main.DAppBrowserPayload
+import io.novafoundation.nova.feature_dapp_impl.presentation.DAppRouter
 import io.novafoundation.nova.feature_deep_linking.presentation.handling.DeepLinkingRouter
 import io.novafoundation.nova.feature_governance_api.presentation.referenda.details.ReferendumDetailsPayload
-import io.novafoundation.nova.feature_governance_impl.presentation.GovernanceRouter
+import io.novafoundation.nova.feature_governance_impl.presentation.referenda.details.ReferendumDetailsFragment
 import io.novafoundation.nova.feature_wallet_api.presentation.model.AssetPayload
 
 class DeepLinkingNavigator(
-    private val accountRouter: AccountRouter,
-    private val assetsRouter: AssetsRouter,
+    navigationHoldersRegistry: NavigationHoldersRegistry,
     private val dAppRouter: DAppRouter,
-    private val governanceRouter: GovernanceRouter
-) : DeepLinkingRouter {
+    private val accountRouter: AccountRouter,
+    private val assetsRouter: AssetsRouter
+) : BaseNavigator(navigationHoldersRegistry), DeepLinkingRouter {
 
     override fun openAssetDetails(payload: AssetPayload) {
-        assetsRouter.openAssetDetails(payload)
+        openSplitScreenWithInstantAction(R.id.action_mainFragment_to_balanceDetailFragment, BalanceDetailFragment.getBundle(payload))
```

Comment:
Why do we use `withInstantAction` for an action that starts from `MainFragment` (action_mainFragment_to_balanceDetailFragment) already? Is there a typo in the action name and it is actually a global one?